Initial documentation for additional tools

mlacouture opened the pull request:
# Title

Documentation for .net core additional tools.

## Summary

This section is meant to compile information about tools in addition to CLI tools.

[Internal Review URL(index.md)](https://review.docs.microsoft.com/en-us/dotnet/core/additional-tools/index?branch=pr-en-us-3871)
[Internal Review URL(wcf-web-service-reference-guide.md)](https://review.docs.microsoft.com/en-us/dotnet/core/additional-tools/wcf-web-service-reference-guide?branch=pr-en-us-3871)
[Internal Review URL(xml-serializer-generator.md)](https://review.docs.microsoft.com/en-us/dotnet/core/additional-tools/xml-serializer-generator?branch=pr-en-us-3871)

## Suggested Reviewers

@zhenlan

mlacouture committed:
Initial documentation for additional tools

mlacouture committed:
fixed image links

mlacouture committed:
updated service endpoint picture

mlacouture committed:
Added warning note

mlacouture committed:
removed text left out accidentally from previous commit

shmao commented on docs/core/additional-tools/WCF-Web-Service-Reference-guide.md:
```diff
@@ -0,0 +1,51 @@
+## Microsoft WCF Web Service Reference Provider
+
+Over the years, many Visual Studio developers have enjoyed the productivity that Add Service Reference provides when their .NET Framework projects needed to access Windows Communication Foundation (WCF) services. The WCF team is excited to introduce the WCF Connected Service for .NET Core and ASP.NET Core, a VS extension providing an experience like the Add Service Reference functionality. This t ...
```
> ... that Add Service Reference provides when their .NET Framework projects needed to ...


It seems to me that either `provides` should be `provided` or `needed` should be `need`.

shmao replied:
> This tool retrieves metadata from a WCF service in the current solution, locally or on a network...

This sentence doesn't sound right.

zhenlan replied:
Yeah, maybe something like "This tool retrieves metadata from a WCF service in the current solution, a WSDL file at the local disk or a web service on a network...".

mlacouture replied:
fixed the 'locally' part, the rest is OK.

shmao commented on docs/core/additional-tools/WCF-Web-Service-Reference-guide.md:
```diff
@@ -0,0 +1,51 @@
+
+### How to use the extension
+
+The WCF Web Service Reference VS connected service extension is applicable to projects created using the templates under Visual C# -> .Net Core, Visual C# -> .Net Standard and Visual C# -> Web -> ASP.NET CORE Web Application projects.
```
> Visual C# -> .Net Standard and Visual C#

Add a `,` before `and`, e.g.,

`Visual C# -> .Net Standard, and Visual C#`

shmao replied:
`ASP.NET CORE` should be `ASP.NET Core`

mlacouture replied:
Fixed the 'CORE' part, the rest is OK - thank you!

mairaw replied:
and .NET is spelled all caps

mlacouture committed:
incorporated feedback

mlacouture committed:
updated text formatting and some text edits

mlacouture committed:
One more formatting fix

mlacouture committed:
renamed file all lowercased

mlacouture committed:
fixed target framework reference

mlacouture committed:
Added entry to the .NET Core Additional Tools index

zhenlan commented on docs/core/additional-tools/wcf_web_service_reference-guide.md:
```diff
@@ -0,0 +1,52 @@
+
+Over the years, many Visual Studio developers have enjoyed the productivity that the [Add Service Reference](https://msdn.microsoft.com/en-us/library/bb386382.aspx) tool provided when their .NET Framework projects needed to access Windows Communication Foundation (WCF) services. The WCF team is excited to introduce the **WCF Web Service Reference** tool, a Visual Studio connected service extensio ...
+
+> [!!IMPORTANT]
```
Two !s may mean it's very important but not sure if it will work ;-)

mlacouture replied:
fixed.

mlacouture committed:
fixed meta char for IMPORTANT tag

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,45 @@
+The following is required for Microsoft.XmlSerializer.Generator to work. You can use command `dotnet --info` to check which versions of .NET Core SDK and runtime you may already have installed.
+
+* [.NET Core SDK 2.0.2 or later](https://www.microsoft.com/net/download/windows)
+* [.NET Core runtime 2.0.3 or later](https://github.com/dotnet/core/blob/master/release-notes/download-archives/2.0.3.md)
```
doesn't the SDK already come with the right runtime versions? Do we need to list the runtime separately?

mlacouture replied:
No, the SDK and the runtime are two different things.

mairaw replied:
I know they are different things. But what I'm asking is, when you install the right SDK, doesn't that come bundled with the right runtime?

shmao replied:
SDK 2.0.2 would install runtime 2.0.2. The tool requires runtime 2.0.3.

SDK 2.0.3 is not released yet.

mairaw replied:
Ok, then it makes sense to list them separately. Thanks @shmao

JRAlexander replied:
@shmao -- Current .NET SDK is 2.1.3
It contains:
    • .NET Core SDK 2.1.3
    **.NET Core Runtime 2.0.4**
    • Runtime Package Store
Will that be ok?

JRAlexander replied:
Current .NET Core SDK version is 2.1.3, which contains .NET Core Runtime 2.0.4.  I've modified to say .NET Core SDK 2.1.3 or later. Ok?

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,45 @@
+ # Using Microsoft Xml Serializer Generator on .NET Core
+
+Like the Xml Serializer Generator (sgen.exe) on desktop, Microsoft.XmlSerializer.Generator NuGet package is the solution for .NET Core and .NET Standard Libraries. It creates an Xml serialization assembly for types contained in an assembly to improve the startup performance of Xml serialization when serializing or de-serializing objects of those types using XmlSerializer. 
```
is the solution for what?

.NET Standard Libraries -> .NET Standard projects

link to the API as shown for index.md

mlacouture replied:
the solution equivalent to sgen.exe on desktop.

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,45 @@
+* [.NET Core SDK 2.0.2 or later](https://www.microsoft.com/net/download/windows)
+* [.NET Core runtime 2.0.3 or later](https://github.com/dotnet/core/blob/master/release-notes/download-archives/2.0.3.md)
+  
+## Instructions
```
can you give a better name? maybe say what they'll be doing in this section?

JRAlexander replied:
I changed to "Use Microsoft XML Serializer Generator in a .NET Core console application"

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,45 @@
+  
+## Instructions
+
+Here are the step by step instructions on how to use Xml Serializer Generator in a .NET Core console application.
```
The following instructions show you how to use...

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,45 @@
+
+Here are the step by step instructions on how to use Xml Serializer Generator in a .NET Core console application.
+
+1. Create a .NET Core console application, e.g. create an app named 'MyApp' with the command,
```
Should tell customers first to open a command prompt window and navigate to where they want to create the project first?

remove ", e.g. create an app named 'MyApp' with the command,"

finish with :

JRAlexander replied:
Yes and to create a directory beforehand.

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
````diff
@@ -0,0 +1,45 @@
+    ```
+    dotnet new console --name MyApp
+    ```
+1. Edit the .csproj and add a reference to the Microsoft.XmlSerializer.Generator package. For example,
````
choose one way of doing it. since you're already on CLI, continue to use the CLI.

You can then tell them they can open the project file and see that the following lines were added.

Something like:

1. Add a reference to the Microsoft.XmlSerializer.Generator package:

`dotnet add package Microsoft.XmlSerializer.Generator -v 1.0.0-preview1-25915-02`

After running the command, the following lines are added to your project file:
 ```xml
    <ItemGroup>
      <DotNetCliToolReference Include="Microsoft.XmlSerializer.Generator" Version="1.0.0-preview1-25915-02" />
    </ItemGroup>
```

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
````diff
@@ -0,0 +1,45 @@
+        public int Value;
+    }
+    ```    
+    Now you can create an `XmlSerializer` for MyClass.    
````
end with :

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
````diff
@@ -0,0 +1,45 @@
+    ```c#
+    var serializer = new System.Xml.Serialization.XmlSerializer(typeof(MyClass));
+    ```    
+1. Build the application by running `dotnet build`. If everything succeeds, an assembly named MyApp.XmlSerializers.dll will be generated in the output folder. You will see warnings in the build output if the tool failed to generate the assembly.
````
everything -> build

will be -> is

remove last sentence

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
````diff
@@ -0,0 +1,45 @@
+    ```    
+1. Build the application by running `dotnet build`. If everything succeeds, an assembly named MyApp.XmlSerializers.dll will be generated in the output folder. You will see warnings in the build output if the tool failed to generate the assembly.
+
+Start the application and it will automatically load and use the pre-generated serializers at runtime.
````
how do you start the application?

it will automatically load and use -> It automatically loads and uses

mlacouture replied:
Start the application by running `dotnet run` from the project folder ..

mairaw commented on docs/toc.md:
```diff
@@ -114,6 +114,7 @@
 ## [Runtime IDentifier catalog](core/rid-catalog.md)
 ## [.NET Core SDK Overview](core/sdk.md)
 ## [.NET Core CLI Tools](core/tools/index.md)
+## [.NET Core Additional Tools](core/additional-tools/index.md)
```
this should be moved to line 149 before the [Porting from .NET Framework](core/porting/index.md) entry

the other two new topics should be under this toc entry with the `###`

mairaw reviewed (changes requested):
Thanks @mlacouture! Finally managed to sit down and review this. Let us know if we can help with the edits.

mlacouture commented on docs/core/additional-tools/wcf_web_service_reference-guide.md:
```diff
@@ -0,0 +1,52 @@
+
+![Connected Services tab](images/WCFCS-ConnectedServicesPage.png)
+
+2. On this page, click **Microsoft WCF Web Service Reference Provider**. This will bring up the **Configure WCF Web Service Reference** wizard:
```
It is a WPF form.  In the context of the connect services framework, it is ok to call it a page which is hosted on the connected services wizard.

mlacouture commented on docs/core/additional-tools/wcf_web_service_reference-guide.md:
```diff
@@ -0,0 +1,52 @@
+
+![Progress window](images/WCFCS-ProgressWindow.png)
+
+This will download metadata from the WCF service, generate the service reference code in a file named `reference.cs`, and add it to your project under the **Connected Services** node. The project file (.csproj) will also be updated with NuGet package references required for your project to compile and run on the target platform.
```
This (the action of clicking Finish). Yes, this is only for C# projects.

mlacouture commented on docs/core/additional-tools/wcf_web_service_reference-guide.md:
```diff
@@ -0,0 +1,52 @@
+* [Release history](https://github.com/dotnet/wcf/blob/master/release-notes/WCF-Web-Service-Reference-notes.md)
+
+
+### Known Issues
```
We can move it to the release notes.

mlacouture commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,45 @@
+
+## Prerequisites
+
+The following is required for Microsoft.XmlSerializer.Generator to work. You can use command `dotnet --info` to check which versions of .NET Core SDK and runtime you may already have installed.
```
This is needed to help people check the required version.  The part 'You can use `dotnet --info` ...' can be moved after the required information.

mairaw replied:
We don't typically say that for any of our tutorials (e.g. https://docs.microsoft.com/en-us/dotnet/core/tutorials/using-on-mac-vs-full-solution#prerequisites) 

Also, if you have a global.json file in your project folder, dotnet --info will give you the sdk you set in the global.json file and not the latest version you have installed.

JRAlexander replied:
.NET Core SDK is currently at 2.1.3, which contains .NET Core runtime 2.0.4. Ok to just say .NET Core SDK 2.1.3 or later?

JRAlexander commented on docs/core/additional-tools/index.md:
```diff
@@ -0,0 +1,19 @@
+title: .NET Core Additional Tools
+description: An overview of the additional tools that support and extend .Net Core functionality.
+author: miguell
+ms.author: miguell
```
Do you want me to change this to johalex instead?

JRAlexander committed:
Revised with recommended changes from review.

Added the recommended changes from the reviewers for this file.

JRAlexander commented on docs/core/additional-tools/wcf_web_service_reference-guide.md:
```diff
@@ -0,0 +1,52 @@
+
+2. On this page, click **Microsoft WCF Web Service Reference Provider**. This will bring up the **Configure WCF Web Service Reference** wizard:
+
+![Service Endpoint tab](images/WCFCS-ServiceEndpointPage.png)
```
I left this as I wasn't sure.

JRAlexander commented on docs/core/additional-tools/wcf_web_service_reference-guide.md:
```diff
@@ -0,0 +1,52 @@
+
+3. You can either click the **Discover** button to initiate a search for services defined in the current solution, enter a service URL in the **Address** box and click the **Go** button to search for services hosted at the specifed address, or click the **Browse** button to select a WSDL file containing the web service metadata information. The services that are found will be displayed in the **S ...
+
+Once a service has been selected you can click the **Next** button to visit the **Data Type Options** and the **Client Options** pages. Alternatively, you can click the **Finish** button to use the default options.
```
Does there need to be a description of the Client Options page?

JRAlexander committed:
Update based on review feedback and rename file to remove spaces

Update based on review feedback and rename wcf_web_service_reference-guide.md to wcf-web-service-reference-guide.md

JRAlexander committed:
Update based on review feedback and format tutorial

Update based on review feedback and format tutorial

JRAlexander commented on docs/core/additional-tools/wcf_web_service_reference-guide.md:
```diff
@@ -0,0 +1,52 @@
+
+![Service Endpoint tab](images/WCFCS-ServiceEndpointPage.png)
+
+3. You can either click the **Discover** button to initiate a search for services defined in the current solution, enter a service URL in the **Address** box and click the **Go** button to search for services hosted at the specifed address, or click the **Browse** button to select a WSDL file containing the web service metadata information. The services that are found will be displayed in the **S ...
```
Modified to bulleted list.

JRAlexander commented on docs/core/additional-tools/wcf_web_service_reference-guide.md:
```diff
@@ -0,0 +1,52 @@
+You can now create instances of the WCF client types generated by the tool and communicate with your web service as desired.
+
+
+### Release Notes
```
I put it in Next Steps

JRAlexander commented on docs/core/additional-tools/wcf_web_service_reference-guide.md:
```diff
@@ -0,0 +1,52 @@
+
+
+### Feedback & Questions
+If you have any questions or feedback, please [open an issue on GitHub](https://github.com/dotnet/wcf/issues/new) so we can follow up with you. You can also review any existing questions or issues from [at this site](https://github.com/dotnet/wcf/issues?utf8=%E2%9C%93&q=is:issue%20label:tooling).
```
Done.

JRAlexander commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,45 @@
+
+The following is required for Microsoft.XmlSerializer.Generator to work. You can use command `dotnet --info` to check which versions of .NET Core SDK and runtime you may already have installed.
+
+* [.NET Core SDK 2.0.2 or later](https://www.microsoft.com/net/download/windows)
```
Also added code editor installation.

JRAlexander committed:
Revised due to feedback

JRAlexander committed:
Revised after review feedback

svick commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
````diff
@@ -0,0 +1,106 @@
+ Type:
+ 
+ ```console
+ dotnet add MyApp package Microsoft.XmlSerializer.Generator -v 1.0.0-preview1-25915-02
````
When I tried this command, it created a normal `<PackageReference>`, not a `<DotNetCliToolReference>`. So, isn't it necessary to edit the csproj by hand?

See also https://github.com/NuGet/Home/issues/4901.

shmao replied:
Yes, `<DotNetCliToolReference>` needs to be added manually.

JRAlexander replied:
It wasn't clear in the original article. Based on the original wording, I assumed that dotnet add created this as well.  Now revised. Thanks!

svick commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,106 @@
+public class MyClass
+ {
+    public int Value;
+ }
```
Nit: the code looks weird with the braces indented.

JRAlexander replied:
Indents are gone.

svick commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,106 @@
+
+### Build and run the application
+
+Still within the *myApp* folder, run the application via [`dotnet run`](../tools/dotnet-run.md)  and it will automatically load and use the pre-generated serializers at runtime.
```
Nit: myApp -> MyApp

svick commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,106 @@
+
+If everything succeeds, an assembly named *MyApp.XmlSerializers.dll* is generated in the output folder. 
+
+Congratulations! you have just:
```
you -> You

svick commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,106 @@
+
+Congratulations! you have just:
+> [!div class="checklist"]
+> * Create a .NET Core app.
```
Create -> Created

svick commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,106 @@
+> * Added a reference to the Microsoft.XmlSerializer.Generator package
+> * Edited your MyApp.csproj to add dependencies.
+> * Added a class and an XmlSerializer
+> * Built and ran the application 
```
Some of the list items are missing a period.

shmao commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,106 @@
+> * How to add a class and an XmlSerializer
+> * How to build and run the application 
+
+Like the [Xml Serializer Generator (sgen.exe)](https://docs.microsoft.com/en-us/dotnet/standard/serialization/xml-serializer-generator-tool-sgen-exe) for .NET Framework, <xref:System.Xml.Serialization.XmlSerializer> NuGet package is the equivalent for .NET Core and .NET Standard projects. It creates an XML serialization assembly for types contained in an assembly to improve the startup performanc ...
```
Right, the doc should talk about [Microsoft.XmlSerializer.Generator](https://www.nuget.org/packages/Microsoft.XmlSerializer.Generator/)

JRAlexander commented on docs/core/additional-tools/index.md:
```diff
@@ -0,0 +1,19 @@
+The WCF Web Service Reference is a Visual Studio connected service provider that made its debut in [Visual Studio 2017 version 15.5](https://www.visualstudio.com/en-us/news/releasenotes/vs2017-relnotes#WCFTools). The WCF Web Service Reference Tool generates a web service reference containing proxy code for a client to easily consume the web service's functionality, similar to the Add Service Refe ...
+
+### [XML Serializer Generator](xmlserializergenerator-instructions.md)
+Like the [Xml Serializer Generator (sgen.exe)](https://docs.microsoft.com/en-us/dotnet/standard/serialization/xml-serializer-generator-tool-sgen-exe) for .NET Framework, <xref:System.Xml.Serialization.XmlSerializer> NuGet package is the solution for .NET Core and .NET Standard Libraries. It creates an XML serialization assembly for types contained in an assembly to improve the startup performance ...
```
Yes, good catch.

JRAlexander commented on docs/core/additional-tools/index.md:
```diff
@@ -0,0 +1,19 @@
+# .NET Core additional tools
+This section compiles a list of tools that support and extend the .NET Core functionality, in addition to the [.NET Core command-line interface (CLI)](..\tools\index.md) tools.
+
+### [WCF Web Service Reference Tool](wcf_web_service_reference-guide.md)
```
The file was renamed.

JRAlexander committed:
Updated file name and removed Nuget package

JRAlexander commented on docs/core/additional-tools/wcf-web-service-reference-guide.md:
```diff
@@ -0,0 +1,77 @@
+* Download metadata from the WCF service 
+* Generate the service reference code in a file named *reference.cs*, and add it to your project under the **Connected Services** node. * Update the project file (.csproj) with  NuGet package references required to compile and run on the target platform.
+
+When these processes complete, you can create generated WCF web service communication client type instances.
```
You are right.  Been thinking about that, not successfully.

JRAlexander committed:
Revised based on review feedback

* Updated Note formatting
* Fixed capitalization
* added space before **Reuse
* Update the project file moved to new line

JRAlexander committed:
Revised based on review

JRAlexander committed:
Removed nuget package

Removed the words "nuget package" from the introduction paragraph.

JRAlexander committed:
Update TOC.md

Corrected and asdded needed entries for TOC

JRAlexander committed:
Redid removing spaces from brackets in code snippet

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,118 @@
+
+### Create a .NET Core console application
+
+Open a command prompt and create a folder named *MyApp*. Navigate to the folder you created and type the following commands:
```
commands -> command

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,118 @@
+
+### Add a reference to the Microsoft.XmlSerializer.Generator package in the MyApp project.
+
+ Type:
```
Type -> Use the dotnet add (link to tool) command to add the reference in your project or something like that

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
````diff
@@ -0,0 +1,118 @@
+dotnet new console
+```
+
+### Add a reference to the Microsoft.XmlSerializer.Generator package in the MyApp project.
````
period in the heading?

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,118 @@
+
+Open your text editor and let's get started! We're still working from the *MyApp* directory we built the app in.
+
+Open *MyApp.csproj* in your text editor.
```
text editor or code editor?

JRAlexander replied:
Code!

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,118 @@
+
+Open *MyApp.csproj* in your text editor.
+
+After running the `dotnet add` command, the following lines are added to your *MyApp.csproj* project file:
```
link to the command page

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
````diff
@@ -0,0 +1,118 @@
+ </ItemGroup>
+ ```
+ 
+ ### Add another ItemGroup section for .NET Core CLI Tool support
````
nit: funny that this header is not showing in blue

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,118 @@
+ 
+ ### Add another ItemGroup section for .NET Core CLI Tool support
+ 
+ Add the following lines after the `ItemGroup` section that we just inspected.
```
end with :

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
````diff
@@ -0,0 +1,118 @@
+ </ItemGroup>
+ ```
+ 
+### Add a class in the application. 
````
period in the heading?

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,118 @@
+* Install your favorite code editor, if you haven't already.
+
+> [!TIP]
+> Need to install a code editor? Try [Visual Studio](https://visualstudio.com/downloads)!
```
the link used by the vsdocs team is this one: https://aka.ms/vsdownload?utm_source=mscom&utm_campaign=msdocs

mairaw commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
````diff
@@ -0,0 +1,118 @@
+ $ dotnet run
+ ```
+> [!NOTE]
+> [`dotnet run`](../tools/dotnet-run.md) calls [`dotnet build`](../tools/dotnet-build.md) to ensure that the build targets have been built, and then calls `dotnet <assembly.dll>` to run the target application.
````
not sure if we should add a note similar to what we have in the end of the following tutorial given that dotnet run is only meant for dev scenarios:
https://docs.microsoft.com/en-us/dotnet/core/tutorials/using-with-xplat-cli

"Note that the commands and steps shown in this tutorial to run your application are used during development time only. Once you're ready to deploy your app, you'll want to take a look at the different deployment strategies for .NET Core apps and the dotnet publish command."

mairaw reviewed (changes requested):
It's coming together nicely. A few extra comments 😃

mlacouture committed:
updated description and progress window image

JRAlexander commented on docs/core/additional-tools/xmlserializergenerator-instructions.md:
```diff
@@ -0,0 +1,118 @@
+> [!div class="checklist"]
+> * How to create a .NET Core app.
+> * How to add a reference to the Microsoft.XmlSerializer.Generator package
+> * How to edit your MyApp.csproj to add dependencies.
```
no periods.

Commit:
Revised based on review feedback

Commit:
Revised production information at bottom

mlacouture committed:
Merge pull request #2 from mlacouture/dev/miguell/dotnetdocs/updates

updated description and progress window image

Commit:
revised metadata

Commit:
Merge branch 'master' of https://github.com/mlacouture/docs

Resyncing changes

JRAlexander commented on docs/core/additional-tools/wcf-web-service-reference-guide.md:
```diff
@@ -0,0 +1,77 @@
+
+3. Select a service.
+
+3a. There are several services search options available within the **Configure WCF Web Service Reference** wizard:
```
I wasn't sure how best to list those substeps, and they are all needed to select the service, which is step 3.

JRAlexander commented on docs/core/additional-tools/wcf-web-service-reference-guide.md:
```diff
@@ -0,0 +1,77 @@
+ * To search for services hosted at a specified address, enter a service URL in the **Address** box and click the **Go** button.
+ * To select a WSDL file containing the web service metadata information, click the **Browse** button. 
+ 
+3b. Select the service from the search results list in the **Services** box. If needed, enter the namespace for the generated code in the corresponding **Namespace** text box.
```
Any of those options will return the list of services.

JRAlexander commented on docs/core/additional-tools/wcf-web-service-reference-guide.md:
```diff
@@ -0,0 +1,77 @@
+While displaying progress, the tool will:
+
+* Download metadata from the WCF service 
+* Generate the service reference code in a file named *reference.cs*, and add it to your project under the **Connected Services** node. * Update the project file (.csproj) with NuGet package references required to compile and run on the target platform.
```
Yes, good catch!

Commit:
TOC changes,feddback revisions,file rename,move images to media

Commit:
Fixed relative link

Commit:
fixed relative links

Commit:
changed media directory and changed filenames to lower case

svick commented on docs/core/additional-tools/xml-serializer-generator.md:
````diff
@@ -0,0 +1,123 @@
+
+```csharp
+var serializer = new System.Xml.Serialization.XmlSerializer(typeof(MyClass));
+```
````
Just creating a serializer is not useful on its own.

Should this tutorial also include some short code that uses it? Or maybe link to another tutorial that explains that?

JRAlexander replied:
Good idea. I linked to some other resources.

svick commented on docs/core/additional-tools/index.md:
```diff
@@ -0,0 +1,19 @@
+The WCF Web Service Reference is a Visual Studio connected service provider that made its debut in [Visual Studio 2017 version 15.5](https://www.visualstudio.com/news/releasenotes/vs2017-relnotes#WCFTools). The WCF Web Service Reference Tool generates a web service reference containing proxy code for a client to easily consume the web service's functionality, similar to the Add Service Reference  ...
+
+### [XML Serializer Generator](xmlserializergenerator-instructions.md)
+Like the [Xml Serializer Generator (sgen.exe)](../standard/serialization/xml-serializer-generator-tool-sgen-exe.md) for .NET Framework, <xref:System.Xml.Serialization.XmlSerializer> is the solution for .NET Core and .NET Standard libraries. It creates an XML serialization assembly for types contained in an assembly to improve the startup performance of XML serialization when serializing or de-ser ...
```
Maybe I was unclear in my previous comment, I think this should say:

> Like the Xml Serializer Generator (sgen.exe) for .NET Framework, the Microsoft.XmlSerializer.Generator NuGet package is the solution for .NET Core and .NET Standard libraries.

Since `XmlSerializer` on its own does not create the serialization assembly.

JRAlexander replied:
Agreed.

svick commented on docs/core/additional-tools/wcf-web-service-reference-guide.md:
```diff
@@ -0,0 +1,78 @@
+
+2. On the **Connected Services** page, click **Microsoft WCF Web Service Reference Provider**. This brings up the **Configure WCF Web Service Reference** wizard:
+
+![Service Endpoint tab](,/media/wcf-web-service-reference-guide/WCFCS-ServiceEndpointPage.png)
```
The comma at the start of the link should be a period.

JRAlexander replied:
Good catch.

svick commented on docs/core/additional-tools/xml-serializer-generator.md:
```diff
@@ -0,0 +1,123 @@
+> * How to add a class and an XmlSerializer
+> * How to build and run the application 
+
+Like the [Xml Serializer Generator (sgen.exe)](https://docs.microsoft.com/en-us/dotnet/standard/serialization/xml-serializer-generator-tool-sgen-exe) for the .NET Framework, <xref:System.Xml.Serialization.XmlSerializer> is the equivalent for .NET Core and .NET Standard projects. It creates an XML serialization assembly for types contained in an assembly to improve the startup performance of XML s ...
```
Similarly to my other comment, I think this should say:

> Like the Xml Serializer Generator (sgen.exe) for the .NET Framework, Microsoft.XmlSerializer.Generator NuGet package is the equivalent for .NET Core and .NET Standard projects.

Commit:
Revising based on review feedback.

Commit:
Added Related resources and revised link

mairaw commented on docs/core/additional-tools/wcf-web-service-reference-guide.md:
```diff
@@ -0,0 +1,78 @@
+﻿--- 
```
remove the hidden character here that is causing the metadata section not to render as a table on preview

mairaw commented on docs/core/additional-tools/index.md:
```diff
@@ -0,0 +1,19 @@
+The WCF Web Service Reference is a Visual Studio connected service provider that made its debut in [Visual Studio 2017 version 15.5](https://www.visualstudio.com/news/releasenotes/vs2017-relnotes#WCFTools). The WCF Web Service Reference Tool generates a web service reference containing proxy code for a client to easily consume the web service's functionality, similar to the Add Service Reference  ...
+
+### [XML Serializer Generator](xmlserializergenerator-instructions.md)
+Like the [Xml Serializer Generator (sgen.exe)](../../standard/serialization/xml-serializer-generator-tool-sgen-exe.md) for .NET Framework, he Microsoft.XmlSerializer.Generator nuget package is the solution for .NET Core and .NET Standard libraries. It creates an XML serialization assembly for types contained in an assembly to improve the startup performance of XML serialization when serializing o ...
```
typically we'd say "for the .NET Framework" with the article

he -> the

nuget  -> NuGet

inline code or create link for Microsoft.XmlSerializer.Generator?

JRAlexander replied:
I created a link to the NuGet package page.

mairaw commented on docs/core/additional-tools/wcf-web-service-reference-guide.md:
```diff
@@ -0,0 +1,78 @@
+---
+# Microsoft WCF Web Service Reference Provider Tool
+
+Over the years, many Visual Studio developers have enjoyed the productivity that the [Add Service Reference](https://docs.microsoft.com/visualstudio/data-tools/how-to-add-update-or-remove-a-wcf-data-service-reference) tool provided when their .NET Framework projects needed to access web services. The **WCF Web Service Reference** tool is a Visual Studio connected service extension that provides a ...
```
remove https://docs.microsoft.com from the link

bold Add Service Reference

mairaw commented on docs/core/additional-tools/wcf-web-service-reference-guide.md:
```diff
@@ -0,0 +1,78 @@
+
+The **Connected Services** page appears as shown in the following image:
+
+![Connected Services tab](./media/wcf-web-service-reference-guide/WCFCS-ConnectedServicesPage.png)
```
images are not rendering. they are still not inside a media folder. also here the image name is still using capital letters, it should be lowercase.

also alt text should be more descriptive,

mairaw commented on docs/core/additional-tools/wcf-web-service-reference-guide.md:
```diff
@@ -0,0 +1,78 @@
+While displaying progress, the tool:
+
+* Downloads metadata from the WCF service. 
+* Generates the service reference code in a file named *reference.cs*, and add it to your project under the **Connected Services** node. 
```
and add -> and adds

mairaw commented on docs/core/additional-tools/wcf-web-service-reference-guide.md:
```diff
@@ -0,0 +1,78 @@
+
+When these processes complete, you can create an instance of the generated WCF client type and invoke the service operations.
+
+## Next Steps
```
sentence case in headings

mairaw commented on docs/core/additional-tools/xml-serializer-generator.md:
````diff
@@ -0,0 +1,131 @@
+var serializer = new System.Xml.Serialization.XmlSerializer(typeof(MyClass));
+```
+
+### Build and run the application
````
should you show the whole file after all the steps are done so people can verify they did right?

JRAlexander replied:
It's a dll.

mairaw replied:
@JRAlexander I was talking about the source code above this step.

mairaw reviewed (approved):
Approving this PR but there are still some comments left to be addressed.

Commit:
Revised based on review comments

Commit:
Revised based on review comments
